File: test/commonASTtests/py-test-files/dog.py
class Animal:
	def __init__(self, _name, _age):
		self.name = _name;
		self.age = _age;

	def birthday(self):
		self.age += 1; 

class Dog(Animal):
	def __init__(self, _name, _age, _breed):
		Animal.__init__(self, _name, _age)
		self.breed = _breed

	def bark(self):
		print self.breed
		print "bark!"	
		

class GermanShepard(Dog):
	def __init__(self, _name, _age):
		Dog.__init__(self, _name, _age, "GermanShepard")

d = Dog("n", 10, "poodle")
d.bark()

File: commonAST/cpp-test-files/classTest/dog.cpp
#include <iostream>

using namespace std;

class Animal{
	public:
		Animal(){
			name = "";
			age = 0;
		}

		Animal(string n, int a){
			name = n;	
			age = a;
		}		

		void birthday(){
			age += 1;
		}

	private:
		string name;
		int age;		
};

class Dog : public Animal{
	public:
		Dog(string n, int a, string b){
			Animal(n, a);
			breed = b;
		}

		Dog(){
			Animal();
			breed = "";
		}

	 	void bark(){
			cout << breed << endl;
			cout << "bark" << endl;
		}
	private:
		string breed;
		
};

class GermanShepard : public Dog{
	public:
		GermanShepard(string n, int a){
			Dog(n,a,"GermanShepard");
		}
};

int main(){
	Dog d("princess", 10, "husky");
	d.bark();

	Dog* e = new Dog("name", 5, "GoldenRetriever");
	e->bark();
}
